articles/ros2_threat_model.md

```diff
@@ -0,0 +1,3421 @@
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Configuration data should be considered as private.</li>
+        <li>Configuration data should acessible by the minimum set of users
+and/or components.</li>
+      </ul>
+    </td>
+    <td><a href="https://doi.org/10.3390/s18051643">Ahmad Yousef, Khalil, Anas
+AlMajali, Salah Ghalyon, Waleed Dweik, and Bassam Mohd. “Analyzing
+Cyber-Physical Threats on Robotic Platforms.” Sensors 18, no. 5 (May 21,
+2018): 1643.</a></td>
+  </tr>
+
+  <tr>
+    <td>A user accidentally misconfigures the robot.</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>15</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="warning">▲</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="warning">▲</td>
+    <td class="danger">✘</td>
+    <td>
+      <ul>
+        <li>Configuration data changes should be reversible.</li>
+        <li>Large change should be applied atomically.</li>
+        <li>Fault monitoring should be able to automatically reset the
+configuration to a safe state if the robot becomes unavailable.</li>
+      </ul>
+    </td>
+    <td> </td>
+  </tr>
+
+  <tr><th colspan="29">Embedded / Software / Data Storage (File
+System)</th></tr>
+
+  <tr>
+    <td>An attacker modifies the robot file system by physically acessing
+it.</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td class="danger">3</td>
+    <td>15</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="danger">✘</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td class="success">✓</td>
+    <td>
+      <ul>
+        <li>Robot filesystem must be encrypted. The key should be stored in a
+secure enclave (TPM).</li>
+        <li>Robot filesystem should be wiped out if the robot perimeter is
+breached.</li>
```

Comment:
What does this actually mean? What is the robot perimeter?

Reply:
I was thinking of perimeter-intrusion in this sense: https://community.zymbit.com/t/using-perimeter-detect/204

Rephrased to make it easier to understand.